keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -1351,56 +1329,129 @@ type PodScheduling struct {
 	// to ensure that the PodScheduling object gets deleted
 	// when no longer needed. Normally the scheduler will delete it.
 	//
+	// Drivers must ignore PodScheduling objects where the owning
+	// pod already got deleted because such objects are orphaned
+	// and will be removed soon.
+	//
 	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
 	metav1.ObjectMeta
 
+	// Spec is set and updated by the scheduler.
```

Comment:
If/when the granular auth goes in, retconning older APIs to use it is going to be controversial or at least difficult without breaking existing users. We can ship RBAC rules that keep everyone working but that doesn't fix it for everyone.

Reply:
I don't think spec and status are a bad fit. "Spec" is what the user wants, "Status" is what they have. But I am open for suggestions if there are better alternatives.

The split is right for authorization: Spec is set by the user, status by the system (including resource driver).

Reply:
I think it is reasonable to use spec for scheduler and status for the responders as an API thing.

...I'm going to try to avoid getting involved in this KEP but on looking at just this it's not easy to guess what circumstance could possibly make this a reasonable dance for the scheduler to have to do.

Reply:
(I'm sure there's a very good reason, but this is still a surprising thing to have in the scheduler path)

Reply:
In contrast to native resources and extended resources, the scheduler is not the one who can decide about resource allocation because it doesn't know what the claim parameters mean. The drivers need to handle that and report the result through the claim status.

Reply:
Agreed with @lavalamp and this is different from how we normally use spec in K8s. But ok for now to move forward.